projects/healthPingValidation.kp/knowledge.md

```diff
@@ -0,0 +1,343 @@
+    keys = map(lambda pair: pair[0], listOfTuples)
+    values = map(lambda pair: pair[1], listOfTuples)
+
+    plt.figure(1)
+    fig = plt.gcf()
+    fig.set_size_inches(15, 7)
+
+    plt.title(title)
+    plt.gca().xaxis.set_major_formatter(mdates.DateFormatter('%m/%d/%Y'))
+    plt.bar(range(len(listOfTuples)), values, align='center', color=inColor)
+    plt.xticks(range(len(listOfTuples)), keys, rotation=90)
+
+```
+
+```python
+data = [(k, v) for k, v in resultDictionary.items()]
+data.sort()
+
+plotlistofTuples(data, "Pings count per day")  
+```
+
+
+![png](images/output_14_0.png)
+
+
+
+```python
+import pandas as pd
+
+pingsPerDaySeries = pd.Series(resultDictionary.values())
+pingsPerDaySeries.describe([.25, .5, .75, .95])
+```
+
+
+
+
+    count       15.000000
+    mean     50329.800000
+    std      20164.844933
+    min       6475.000000
+    25%      37564.500000
+    50%      56195.000000
+    75%      59652.500000
+    95%      76980.000000
+    max      80578.000000
+    dtype: float64
+
+
+
+# Compute how many clients are reporting
+
+
+```python
+def getClients(pings):
+    clients = pings.map(lambda ping: ping["clientId"]).distinct()
+    return clients
+```
+
+```python
+clientsCount = getClients(cachedData).count()
+print "Clients number = " + str(clientsCount)
+```
+    Clients number = 67028
+
+
+# Compute average number of pings per client per day
+
+This include showing diagrams and printings stats
+
+
+
+```python
+from collections import Counter
+
+def getAvgPerDate(iterable):
+    aggregare = Counter(iterable)
+    result = sum(aggregare.values()) * 1.0 / len(aggregare)
+    return result
+    
+def pingsPerClientPerDay(pings):
+    idDateRDD = pings.map(lambda ping: (ping["clientId"], ping["meta/submissionDate"])).groupByKey()
+    return idDateRDD.map(lambda pair: (pair[0], getAvgPerDate(pair[1]))).collect()
+
+resultDistributionDict = pingsPerClientPerDay(cachedData)
+```
+
+```python
+values = map(lambda p: p[1], resultDistributionDict)
+
+plt.title("Average number of pings per client per day")
+plt.plot(sorted(values))
+
+pingsPerDaySeries = pd.Series(values)
+pingsPerDaySeries.describe([.25, .5, .75, .95])
+
+```
+
+
+
+
+    count    67028.000000
+    mean         3.041343
+    std         12.124944
+    min          1.000000
+    25%          1.333333
+    50%          2.000000
+    75%          3.222222
+    95%          7.090909
+    max       1278.000000
+    dtype: float64
+
+
+
+
+
+![png](images/output_21_1.png)
+
+
+# Although pings could be sent only 1 time in a hour (at max 24 times per day), some clients submitted more than 24 pings. 
+Half of this number could be described because of shutdown ping (TelemetrySend save this ping on shutdown and submit as pending in the next session, which could lead to loats of ping)
+
+# Daily active Health ping clients against DAU
+
+
+```python
+DAU = [('20170730', 28097), ('20170728', 33078), ('20170802', 36030), ('20170807', 40198), ('20170808', 37232),
```

Comment:
I would like to see ratios for beta, but for beta putting DAU numbers in public repos is probably not a good idea.
Lets check if we can just:
- load the [client_count dataset](https://docs.telemetry.mozilla.org/datasets/batch_view/client_count/reference.html) from the notebook
- with that avoid showing numbers by only displaying ratios of "clients reporting health pings" vs. "DAU"